website/addons/evernote/model.py

```diff
@@ -0,0 +1,127 @@
+
+from website.addons.base import AddonOAuthUserSettingsBase, AddonOAuthNodeSettingsBase
+from website.addons.base import StorageAddonBase
+from website.addons.evernote import (settings, utils)
+from website.addons.evernote.serializer import EvernoteSerializer
+from website.oauth.models import (ExternalProvider, OAUTH1)
+
+from modularodm import fields
+
+import logging
+logger = logging.getLogger(__name__)
+
+class Evernote(ExternalProvider):
+    """
+    First cut at the Evernote provider
+
+    """
+    name = 'Evernote'
+    short_name = 'evernote'
+
+    # Evernote has a sandbox and a production service
+    # with different endpoints
+
+    client_id = settings.EVERNOTE_CLIENT_ID
+    client_secret = settings.EVERNOTE_CLIENT_SECRET
+    _sandbox = settings.EVERNOTE_SANDBOX
+
+    BASE_URL = "https://www.evernote.com" if not _sandbox  \
+           else "https://sandbox.evernote.com"
+
+    auth_url_base = '{}/OAuth.action'.format(BASE_URL)
+    request_token_url = '{}/oauth'.format(BASE_URL)
+    callback_url = '{}/oauth'.format(BASE_URL)
+
+    _oauth_version = OAUTH1
+
+    def handle_callback(self, response):
+        """View called when the Oauth flow is completed. Adds a new BoxUserSettings
+        record to the user and saves the user's access token and account info.
+        """
+
+        client = utils.get_evernote_client(token=response.get('oauth_token'))
+
+        userStore = client.get_user_store()
+        user = userStore.getUser()
+
+        return {
+            'provider_id': user.id,  # or user.username
+            'display_name': user.name,
+            'profile_url': ''
+        }
+
+class EvernoteUserSettings(AddonOAuthUserSettingsBase):
+    oauth_provider = Evernote
+    serializer = EvernoteSerializer
+
+
+class EvernoteNodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
+    oauth_provider = Evernote
+    serializer = EvernoteSerializer
+
+    folder_id = fields.StringField(default=None)
+    folder_name = fields.StringField()
+    folder_path = fields.StringField()
+
+    # to hold data about current Notebook
+    _folder_data = None
```

Comment:
Are you sure this line is unnecessary? Looking at the [box add-on code](https://github.com/CenterForOpenScience/osf.io/blob/095b16bccf4d929903555a5234bc1321d85621ba/website/addons/box/model.py#L89), I see:

```Python
_folder_data = None
```

I was mimicking this code and thinking that if this line is need for `box`, it would be needed for `evernote`.